eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -0,0 +1,151 @@
+steps:
+  - task: DotNetCoreInstaller@2
+    displayName: "Use .NET Core sdk $(DotNetCoreSDKVersion)"
+    inputs:
+      version: $(DotNetCoreSDKVersion)
+  - task: UsePythonVersion@0
+    displayName: 'Use Python 3.6'
+    inputs:
+      versionSpec: '3.6'
+  - template: eng/pipelines/templates/scripts/replace-relative-links.yml@azure-sdk-tools
+    parameters:
+      TargetFolder: '.'
+      RootFolder: '.'
+      BuildSHA: $(Build.SourceVersion)
+      RepoId: 'Azure/azure-sdk-for-net'
+  - pwsh: |
+      # Download and Extract or restore Packages required for Doc Generation
+      Write-Host "Download and Extract mdoc to Build.BinariesDirectory/mdoc"
+      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+      -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "./mdoc/"
+
+      Write-Host "Download and Extract docfx to Build.BinariesDirectory/docfx"
+      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+      -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+
+      Write-Host "Restore eng/docgeneration/assets/docgen.csproj, to get ECMA2Yml and popimport"
+      dotnet restore "$(Build.SourcesDirectory)/eng/docgeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$(Build.BinariesDirectory)
+    workingDirectory: $(Build.BinariesDirectory)
+    displayName: Download and Extract Required Software
+
+  - ${{ if ne(variables['Build.DefinitionName'], 'net - client - ci')}}:
+    - ${{ each artifact in parameters.Artifacts }}:
+      - pwsh: |
```

Comment:
I like that you made this all PS now but as part of the clean-up it would be good to pull it out into a script so we can easily run and debug it locally as well.

Reply:
I will do that as part on my Doc CleanUp Issue